components/bl-autocomplete-component/src/options/index.js

```diff
@@ -30,15 +32,24 @@ export function Options(props) {
     <div
       ref={ optionsContainerRef }
       className={ cn('options', { ['options__placement-top']: optionsPlacement === 'top' }) }>
-      { optionList.map(item => (
-        <Option
-          key={ item.value }
-          item={ item }
-          setInputValue={ setInputValue }
-          setIsOptionsOpen={ setIsOptionsOpen }
-          setAutocompleteValue={ setAutocompleteValue }
-          onChange={ onChange }
-        />
+      { optionsList.map(item => (
+        hasGroup
+          ? <GroupedOptions
+              key={ item.value }
+              item={ item }
+              setInputValue={ setInputValue }
+              setIsOptionsOpen={ setIsOptionsOpen }
+              setAutocompleteValue={ setAutocompleteValue }
+              onChange={ onChange }
+            />
+          : <Option
+              key={ item.value }
+              item={ item }
+              setInputValue={ setInputValue }
+              setIsOptionsOpen={ setIsOptionsOpen }
+              setAutocompleteValue={ setAutocompleteValue }
+              onChange={ onChange }
+            />
```

Comment:
const OptionComponent = useMemo(() => hasGroup ? GroupedOptions : Option, [hasGroup])

{ optionsList.map(item => ( <OptionComponent .../>) }

Reply:
agreed, but do we really need useMemo here? reference on `GroupedOptions` or `Option` is static, so we can omit useMemo here, am I correct?

Reply:
Yes, my mistake.